src/Sentry.NLog/SentryTarget.cs

```diff
@@ -0,0 +1,249 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+using NLog;
+using NLog.Config;
+using NLog.Targets;
+
+using Sentry.Extensibility;
+using Sentry.Infrastructure;
+using Sentry.Protocol;
+using Sentry.Reflection;
+
+namespace Sentry.NLog
+{
+    [Target("Sentry")]
+    public sealed class SentryTarget : TargetWithContext
+    {
+        private readonly Func<IHub> _hubAccessor;
+        private readonly ISystemClock _clock;
+        private IDisposable _sdkDisposable;
+
+        internal static readonly (string Name, string Version) NameAndVersion = typeof(SentryTarget).Assembly.GetNameAndVersion();
+
+        private static readonly string ProtocolPackageName = "nuget:" + NameAndVersion.Name;
+
+        public SentryTarget() : this(new SentryNLogOptions())
+        {
+        }
+
+        public SentryTarget(SentryNLogOptions options)
+            : this(
+                options,
+                () => HubAdapter.Instance,
+                sdkInstance: null,
+                SystemClock.Clock)
+        {
+        }
+
+        internal SentryTarget(SentryNLogOptions options, Func<IHub> hubAccessor, IDisposable sdkInstance, ISystemClock clock)
+        {
+            Debug.Assert(options != null);
+            Debug.Assert(hubAccessor != null);
+            Debug.Assert(clock != null);
+
+            // Overrides default layout. Still will be explicitly overwritten if manually configured in the
+            // NLog.config file.
+            Layout = "${message}";
+
+            Options = options;
+            _hubAccessor = hubAccessor;
+            _clock = clock;
+
+            if (sdkInstance != null)
+            {
+                _sdkDisposable = sdkInstance;
+            }
+        }
+
+        public SentryNLogOptions Options { get; }
+
+        #region Xml property setters
+
+        /// <summary>
+        /// Add any desired additional tags that will be sent with every message.
+        /// </summary>
+        [ArrayParameter(typeof(TargetPropertyWithContext), "tag")]
+        public IList<TargetPropertyWithContext> Tags => Options.Tags;
+
+        /// <summary>
+        /// The Data Source Name of a given project in Sentry.
+        /// </summary>
+        [RequiredParameter]
+        public string Dsn
```

Comment:
Like the forwarding of config `Options` as properties on the NLog-target. This makes this NLog-target behave like other NLog-targets. But why are only some of the `Options` forwarded?

Reply:
The short answer is that I didn't feel like re-writing all of the properties again on the target, but these more common and/or required properties felt important enough to justify "duplicating" the code. But if the consensus is that the properties should all be forwarded, I'll gladly update it to do so.

Reply:
Agreed. One can always initialize with `SentrySdk.Init` and avoid setting the `Dsn` via the `NLog` integration. The `NLog` integration then should use the client already available in the root scope. This is how `log4net` and `Serilog` work.